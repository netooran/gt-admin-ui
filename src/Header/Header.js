import styled from 'styled-components';
import { Moon, Sun } from '@styled-icons/boxicons-solid';
import { useDarkMode } from 'hooks';

const StyledHeader = styled.header`
  display: flex;
  justify-content: flex-end;
  padding: 2rem 3rem;
  height: 2.2rem;

  svg {
    width: 2rem;
    cursor: pointer;

    :hover {
      width: 2.2rem;
    }
  }
`;

const Header = () => {
  const [isDarkModeOn, toggleDarkMode] = useDarkMode();

  const Icon = isDarkModeOn ? Sun : Moon;

  return (
    <StyledHeader>
      <Icon
        fill={isDarkModeOn ? '#d89211' : 'inherit'}
        onClick={toggleDarkMode}
      />
    </StyledHeader>
  );
};

export default Header;
