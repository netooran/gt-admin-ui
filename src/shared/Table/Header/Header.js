import styled from 'styled-components';
import PropTypes from 'prop-types';
import Input from 'shared/Input/Input';

const Wrapper = styled.div`
  padding: 1rem;
`;

const Header = ({ options, search, searchPlaceholderText = 'Search...' }) => {
  const { searchable } = options;

  return (
    <Wrapper>
      {searchable && (
        <Input
          type="text"
          role="search"
          aria-label="Search"
          placeholder={searchPlaceholderText}
          onChange={search}
          padding=".25rem 1rem"
        />
      )}
    </Wrapper>
  );
};

Header.propTypes = {
  searchable: PropTypes.bool,
  searchPlaceholderText: PropTypes.string,
  search: PropTypes.func,
  options: PropTypes.shape({
    searchable: PropTypes.bool,
  }),
};

export default Header;
