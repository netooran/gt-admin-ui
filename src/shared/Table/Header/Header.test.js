import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Header from './Header';

test('renders header with search', () => {
  const props = {
    options: { searchable: true },
    search: jest.fn(),
    searchPlaceholderText: 'Search by name or email',
  };

  const { container, getByRole } = render(<Header {...props} />);

  expect(container).toMatchSnapshot();

  userEvent.type(getByRole('search'), 'Ale');
  expect(getByRole('search')).toHaveValue('Ale');
  expect(props.search).toHaveBeenLastCalledWith('Ale');
});

test('renders header without search', () => {
  const props = { options: { searchable: false } };
  const { container } = render(<Header {...props} />);
  expect(container).toMatchSnapshot();
});
