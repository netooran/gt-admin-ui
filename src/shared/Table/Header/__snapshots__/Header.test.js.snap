// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header with search 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border: 1px solid var(--border-color);
  border-radius: 0.2rem;
  background-color: var(--control-bg-color);
  min-width: 250px;
  padding: .25rem 1rem;
}

.c3:focus-within {
  border: 1px solid var(--control-focus-color);
}

.c3 svg {
  fill: var(--danger-color);
  width: 1.2rem;
  padding: 0 0.5rem;
}

.c4 {
  border-style: none;
  border-radius: 0.2rem;
  padding: 0.5rem;
  width: 100%;
  background-color: inherit;
  color: inherit;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  color: inherit;
  background-color: transparent;
  pointer-events: none;
}

.c0 {
  padding: 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      aria-label="Search"
      class="c1 c2 c3"
      placeholder="Search by name or email"
    >
      <input
        aria-label="Search"
        class="c4"
        placeholder="Search by name or email"
        role="search"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`renders header without search 1`] = `
.c0 {
  padding: 1rem;
}

<div>
  <div
    class="c0"
  />
</div>
`;
